src/java.base/share/classes/java/util/IdentityHashMap.java

```diff
@@ -1192,7 +1160,7 @@ public boolean contains(Object o) {
         }
         public boolean remove(Object o) {
             return o instanceof Entry<?, ?> entry
-                    && removeMapping(entry.getKey(), entry.getValue());
+                    && IdentityHashMap.this.remove(entry.getKey(), entry.getValue());
```

Comment:
I would prefer to keep the internal `removeMapping` method and have other methods call it, instead of calling a public method. In particular the `EntrySet.remove()` method here should call an internal method to perform the actual removal instead of calling the public `remove()` method, since that potentially exposes the "self-use" to subclasses.